terraform/api.tf

```diff
@@ -1,14 +1,55 @@
+resource "aws_s3_bucket" "api" {
+  bucket = var.api_sources_bucket
+  acl    = "private"
+  versioning {
+    enabled = true
+  }
+}
+
+resource "aws_s3_bucket_policy" "api_lambda" {
+  bucket = aws_s3_bucket.api.id
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect   = "Allow"
+        Action   = "s3:GetObject"
+        Resource = "${aws_s3_bucket.api.arn}/*"
+        Principal = {
+          AWS = aws_iam_role.lambda.arn
+        }
+      },
+      {
+        Effect   = "Allow"
+        Action   = "s3:ListBucket"
+        Resource = aws_s3_bucket.api.arn
+        Principal = {
+          AWS = aws_iam_role.lambda.arn
+        }
+      }
+    ]
+  })
+}
+
+resource "aws_s3_bucket_object" "api_lambda" {
+  bucket = aws_s3_bucket.api.bucket
+  key    = "api-lambda-s3"
+  source = "./build/empty_lambda.zip"
```

Comment:
How about specifying api.zip, supposedly built just before running terraform?

Reply:
ah I tried that, says error file not found, troubleshooting that now